xs2a-adapter-service-impl/src/main/java/de/adorsys/xs2a/adapter/impl/http/wiremock/WiremockStubDifferenceDetectingInterceptor.java

```diff
@@ -183,12 +184,43 @@ public WiremockStubDifferenceDetectingInterceptor(Aspsp aspsp) {
         return analyzePayloadStructure(requestBody, builder.body(), PayloadType.REQUEST);
     }
 
-    private Optional<String> analyzeRequestHeaders(Request.Builder builder, Map<String, Object> requestHeaders) {
-        if (requestHeaders != null && !builder.headers().keySet().containsAll(requestHeaders.keySet())) {
-            log.warn("{} stub headers are different from the request", aspsp.getAdapterId());
-            return Optional.of("request-headers");
+    private Optional<String> analyzeRequestHeaders(Request.Builder builder, Map<String, Object> stubHeaders) {
+        Map<String, String> currentHeaders = builder.headers();
+        Sets.SetView<String> headers = Sets.intersection(WiremockSupportedHeader.set(), currentHeaders.keySet());
+        Sets.SetView<String> presentHeaders = Sets.intersection(headers, stubHeaders.keySet());
+        Sets.SetView<String> requestDiff = Sets.difference(headers, stubHeaders.keySet());
+        Sets.SetView<String> stubDiff = Sets.difference(stubHeaders.keySet(), headers);
+        StringBuilder differences = new StringBuilder();
+        presentHeaders.forEach(key -> {
+            WiremockSupportedHeader supportedHeader = WiremockSupportedHeader.resolve(key).get();
+            if (!supportedHeader.isEqual(getStubRequestHeader(stubHeaders, key), currentHeaders.get(key))) {
```

Comment:
`isEqual` always returns _true_, except for Content-Type header

Reply:
Is it an expected behavior for `isEqual` always to return _true_, except for cases where the value must be verified?

Reply:
yes. for now, we compare the value of the header only for Content-Type header. For other headers, the comparison is ignored.